### PR TITLE
#EW-2442 EW-2443 User creation endpoint added

### PR DESCRIPTION
**Pull request by Magnico**
Changes made:
- Created endpoint for creating a user
- Created endpoint for retrieving all users
- Added validations for email and username syntax
- Added validations for unique email and username
- Created corresponding tests

### CODE REVIEW

File: app/routes/user.py
Comment (jaagomezmu) [inline app/routes/user.py:1]: According to the Python code style guidelines known as [PEP 8](https://peps.python.org/pep-0008/#imports), imports in a Python script should be grouped into three different sections, in the following order:

    a. Imports of standard Python modules.
    b. Third-party module imports.
    c. Imports of local or project-specific modules.

**Review by jaagomezmu — APPROVED** [review]
Looks good, 
In a next order I would order the imports

File: app/routes/user.py
Comment (ejariza-evowill) [reply app/routes/user.py:1]: Please follow these guidelines and implement the corresponding changes @Magnico

Comment (ejariza-evowill) [reply app/routes/user.py:1]: .

**Review by ejariza-evowill — CHANGES_REQUESTED** [review]

**Review by jaagomezmu — APPROVED** [review]
looks good!

**Pull request merged**